test/e2e-layering/onclusterbuild_test.go

```diff
@@ -225,3 +228,91 @@ func prepareForTest(t *testing.T, cs *framework.ClientSet, testOpts onClusterBui
 	_, err = helpers.WaitForRenderedConfig(t, cs, testOpts.poolName, "00-worker")
 	require.NoError(t, err)
 }
+
+func TestSSHKeyAndPasswordForOSBuilder(t *testing.T) {
+	cs := framework.NewClientSet("")
+
+	// label random node from pool, get the node
+	unlabelFunc := helpers.LabelRandomNodeFromPool(t, cs, "worker", "node-role.kubernetes.io/layered")
+	osNode := helpers.GetSingleNodeByRole(t, cs, layeredMCPName)
+
+	// prepare for on cluster build test
+	prepareForTest(t, cs, onClusterBuildTestOpts{
+		imageBuilderType: build.OpenshiftImageBuilder,
```

Comment:
suggestion (non-blocking): In the future, I think it would make sense to provide a MachineConfig to the `prepareForTest` and `runOnClusterBuildTest` functions so that it can be created and included in the original image build process.